source/common/buffer/watermark_buffer.h

```diff
@@ -40,20 +45,25 @@ class WatermarkBuffer : public OwnedImpl {
   uint32_t highWatermark() const { return high_watermark_; }
 
 private:
-  void checkHighWatermark();
+  void checkHighAndOverflowWatermarks();
   void checkLowWatermark();
 
   std::function<void()> below_low_watermark_;
   std::function<void()> above_high_watermark_;
+  std::function<void()> above_overflow_watermark_;
 
   // Used for enforcing buffer limits (off by default). If these are set to non-zero by a call to
   // setWatermarks() the watermark callbacks will be called as described above.
   uint32_t high_watermark_{0};
   uint32_t low_watermark_{0};
+  uint32_t overflow_watermark_{0};
+  // Multiplier of high_watermark_ that determines the overflow_watermark_ value
+  uint32_t overflow_watermark_multiplier_{0};
```

Comment:
Do we actually need this member variable now that it comes from runtime w/ default? Can we just load this during the set call?

Reply:
Indeed, it is redundant. I'll remove it.